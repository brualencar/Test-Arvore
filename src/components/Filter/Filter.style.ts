import { styled } from 'styled-components';

export const Wrapper = styled.div`
  font-family: Inter;
`;

export const MainTitle = styled.p`
  color: ${({ theme }) => theme.colors.text.neutral.strong};
  font-size: ${({ theme }) => theme.font.size.m5};
  font-weight: ${({ theme }) => theme.font.weight.bold};
`;

export const Title = styled.p`
  color: ${({ theme }) => theme.colors.text.neutral.medium};
  font-size: ${({ theme }) => theme.font.size.m4};
  font-weight: ${({ theme }) => theme.font.weight.bold};
`;

export const Filter = styled.div`
  & > ul {
    list-style-type: none;
    padding: 0;
    margin-bottom: ${({ theme }) => theme.spacing.km8};
  }
  & > ul > li {
    margin-bottom: ${({ theme }) => theme.spacing.km3};
  }
  & > ul > li > span {
    color: ${({ theme }) => theme.colors.text.neutral.primary};
  }
`;
