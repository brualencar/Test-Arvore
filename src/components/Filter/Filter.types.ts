export interface InputEventProps {
  value: string;
}
