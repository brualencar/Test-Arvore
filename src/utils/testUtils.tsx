import React, { PropsWithChildren } from 'react';

import { ThemeProvider } from 'styled-components';

import { render as rtlRender, RenderOptions } from '@testing-library/react';

import { theme } from '../assets/styles/globalStyle';

function render(ui: React.ReactElement, renderOptions?: RenderOptions) {
  function Wrapper({ children }: PropsWithChildren<object>): JSX.Element {
    return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { render };
