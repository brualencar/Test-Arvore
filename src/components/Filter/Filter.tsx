import React from 'react';

import { ItemProps, Root } from 'pages/SearchResult/SearchResult';

import { Button } from '../Button';
import * as Styled from './Filter.style';

type Item = {
  text: string;
  value: string | number;
  complexValue?: { from: number; to?: number };
};

export type OnFilterChangeParams = {
  [key: string]: { value: Item['value']; complexValue: Item['complexValue'] };
};

export interface IFilterProps {
  mainTitle: string;
  filters: Root;
  dispatch: React.Dispatch<any>;
  hasSelectedFilters: boolean;
  resetFilters: () => void;
}

function Filter({
  mainTitle,
  filters,
  dispatch,
  hasSelectedFilters,
  resetFilters,
}: IFilterProps) {
  const handleToggleCheckbox = (itemIndex: any, filterType: string) => {
    dispatch({ type: 'TOGGLE_CHECKBOX', itemIndex, filterType });
  };

  return (
    <Styled.Wrapper>
      <Styled.MainTitle>{mainTitle}</Styled.MainTitle>
      {hasSelectedFilters && (
        <Button
          isUppercase
          text='Limpar Filtro'
          variant='secondary'
          onClick={resetFilters}
        />
      )}
      {Object.entries(filters).map(([filterType, category]) => (
        <div key={filterType}>
          <Styled.Title>{category.title}</Styled.Title>
          <Styled.Filter>
            <ul>
              {category?.items?.map((item: ItemProps, index: any) => (
                <li key={item.id}>
                  <input
                    name={item.label}
                    type='checkbox'
                    checked={item.checked}
                    onChange={() => handleToggleCheckbox(index, filterType)}
                  />
                  <span>{item.label}</span>
                </li>
              ))}
            </ul>
          </Styled.Filter>
        </div>
      ))}
    </Styled.Wrapper>
  );
}

export default Filter;
