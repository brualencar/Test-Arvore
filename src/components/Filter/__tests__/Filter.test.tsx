import React from 'react';

import { render } from '../../../utils/testUtils';
import Filter from '../Filter';

describe('Filter', () => {
  it('should render snapshot', () => {
    const { container } = render(
      <Filter
        mainTitle='Title'
        clearFilter={() => undefined}
        onFilterChange={() => undefined}
        filters={[
          {
            title: 'Preço',
            id: 'price',
            items: [
              {
                text: 'de R$0 até R$30',
                value: '30',
                complexValue: {
                  from: 0,
                  to: 30,
                },
              },
              {
                text: 'de R$31 até R$50',
                value: '50',
                complexValue: {
                  from: 31,
                  to: 50,
                },
              },
              {
                text: 'de R$51 até R$100',
                value: '100',
                complexValue: {
                  from: 51,
                  to: 100,
                },
              },
              {
                text: 'Mais de R$100',
                value: '101',
                complexValue: {
                  from: 101,
                },
              },
            ],
          },
        ]}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
