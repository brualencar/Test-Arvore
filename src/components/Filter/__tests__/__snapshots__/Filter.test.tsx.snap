// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should render snapshot 1`] = `
<div>
  <div
    class="sc-gtcAbF jfXzsC"
  >
    <p
      class="sc-dkXrac lbHbYC"
    >
      Title
    </p>
    <div>
      <p
        class="sc-hKVryf kiBiQw"
      >
        Pre√ßo
      </p>
      <div
        class="sc-eCQiwC cPAvbM"
      >
        <ul>
          <li>
            <input
              type="checkbox"
            />
            <span />
          </li>
          <li>
            <input
              type="checkbox"
            />
            <span />
          </li>
          <li>
            <input
              type="checkbox"
            />
            <span />
          </li>
          <li>
            <input
              type="checkbox"
            />
            <span />
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
